src/Collection/CollectionTrait.php

```diff
@@ -291,8 +291,8 @@ public function indexBy($path): CollectionInterface
             $pathValue = $callback($value);
             if ($pathValue === null) {
                 throw new InvalidArgumentException(
-                    'Cannot use a nonexistent path or null value. ' .
-                    'Use a callback to provide default values if necessary.'
+                    'Cannot index by path that does not exist or contains a null value. ' .
+                    'Use a callback to return a default value.',
```

Comment:
```suggestion
                    'Use a callback to return a default value.'
```
The trailing commas are an error in older php

Reply:
Apparently removing a parameter is very difficult work.